ActionSpeedX/ArmorPatcher.cs

```diff
@@ -64,72 +64,81 @@ public ArmorPatcher(IPatcherState<ISkyrimMod, ISkyrimModGetter> state, ActionSpe
+                            if (kw.EditorID != null && this.materialRanks.ContainsKey(kw.EditorID))
+                            {
+                                int rank = this.materialRanks[kw.EditorID];
+                                if (rank > tier) tier = rank;
+                            }
                         }
                     }
-                }
-               
-                if (tier < 0)
-                {
-                    Console.WriteLine($"No recognized material type for {armor.EditorID}. Skipping. ");
-                    // tier = 1;
-                    continue;
-                }
-                else
-                {
-                    tier--; //correct array access
-                }
-                string slot;
 
-                if (armor.Keywords.Contains(Skyrim.Keyword.ArmorBoots)) slot = BOOTS;
-                else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorCuirass)) slot = CUIRASS;
-                else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorGauntlets)) slot = GAUNTLETS;
-                else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorHelmet)) slot = HELMET;
-                else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorShield)) slot = SHIELD;
-                else
+                    if (tier < 0)
+                    {
+                        Console.WriteLine($"No recognized material type for {armor.EditorID}. Skipping. ");
+                        // tier = 1;
+                        continue;
+                    }
+                    else
+                    {
+                        tier--; //correct array access
+                    }
+                    string slot;
+
+                    if (armor.Keywords.Contains(Skyrim.Keyword.ArmorBoots)) slot = BOOTS;
+                    else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorCuirass)) slot = CUIRASS;
+                    else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorGauntlets)) slot = GAUNTLETS;
+                    else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorHelmet)) slot = HELMET;
+                    else if (armor.Keywords.Contains(Skyrim.Keyword.ArmorShield)) slot = SHIELD;
+                    else
+                    {
+                        Console.WriteLine("No matching equip slot for " + armor.EditorID);
+                        continue;
+                    }
+                    var nw = state.PatchMod.Armors.GetOrAddAsOverride(armor);
+                    nw.Keywords?.Add(armorKeysMap[slot][tier]);
+                    if (this.settings.Descriptions) PatchArmorDescription(nw, armorType, slot, tier);
+                } catch (Exception e)
+                
                 {
-                    Console.WriteLine("No matching equip slot for " + armor.EditorID);
-                    continue;
+                    Console.WriteLine($"Error processing armor record {armor.EditorID}  from {armor.FormKey.ModKey.FileName} : {e.Message}. Skipping");

```

Comment:
Mutagen's got a helper setup for this:

`throw RecordException.Factory("Some message", armor, ex);`

Will print a nice little setup with all the goodies

Reply:
done